src/IR/DConToFunc.hs

```diff
@@ -114,9 +114,89 @@ createArityEnv defs = M.fromList $ concatMap arities defs
 dConToFunc :: I.Program Poly.Type -> Compiler.Pass (I.Program Poly.Type)
 dConToFunc p@I.Program { I.programDefs = defs, I.typeDefs = tDefs } =
   runArityFn (createArityEnv tDefs) $ do
-    defs' <- everywhereM (mkM dataToApp) defs
-    return p { I.programDefs = defs' ++ concat (createFuncs <$> tDefs) }
+    defs'' <- defs' -- user defined functions
+    return p { I.programDefs = tDefs' ++ defs'' } -- constructor funcs ++ user funcs
  where
+  tDefs' = concat (createFuncs <$> tDefs)
+  -- ^ top-level constructor functions
```

Comment:
You should only really use `-- ^` for things like data constructors, where you document the item inline